src/main/java/homework_3/ImmutableClass.java

```diff
@@ -0,0 +1,46 @@
+package homework_3;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public final class ImmutableClass {
```

Comment:
Your class is not immutable. I can pass List and then change it, because when you get/set mutable field, you don't make deep copy, you pass the same object (in constructor and in getNew)